store/chat_test.go

```diff
@@ -0,0 +1,175 @@
+package store
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/dgraph-io/badger/v3"
+	"github.com/pastelnetwork/go-commons/log"
+)
+
+var (
+	chatDB *ChatDB
+)
+
+func TestMain(m *testing.M) {
+	tmpDir, err := ioutil.TempDir("", "chat")
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("Created temporary directory", tmpDir)
+	chatDB = NewChatDB()
+	cfg := NewConfig(tmpDir)
+	if err := chatDB.Start(cfg, log.NewLogger()); err != nil {
+		panic(err)
+	}
+	code := m.Run()
+	fmt.Println("Deleting", tmpDir)
+	os.RemoveAll(tmpDir)
+	os.Exit(code)
+}
+
+func TestChatDB_Set(t *testing.T) {
+	type args struct {
+		key   []byte
+		value []byte
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Storing a new value into DB [hello-world]: OK",
+			args: args{
+				key:   []byte("hello"),
+				value: []byte("world"),
+			},
+			wantErr: false,
+		},
+		{
+			name: "Storing a new value into DB [abc-def]: OK",
+			args: args{
+				key:   []byte("abc"),
+				value: []byte("def"),
+			},
+			wantErr: false,
+		},
+		{
+			name: "Storing a new value into DB [___-&^%]: OK",
+			args: args{
+				key:   []byte("___"),
+				value: []byte("&^%"),
+			},
+			wantErr: false,
+		},
+		{
+			name: "Storing a new value into DB [-]: Error",
+			args: args{
+				key:   []byte(""),
+				value: []byte(""),
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := chatDB.Set(tt.args.key, tt.args.value); (err != nil) != tt.wantErr {
+				t.Errorf("Set() error = %v, wantErr %v", err, tt.wantErr)
```

Comment:
I didn't have a chance to write *go-guide*, but I strongly recommend to use https://github.com/stretchr/testify

https://blockchain-devel.slack.com/archives/C01SLR99SAK/p1617979606143500